documentation/src/main/asciidoc/topics/con_cache_store_jdbc_connection_factories.adoc

```diff
@@ -0,0 +1,65 @@
+[id='jdbc-connection-factories_{context}']
+= JDBC connection factories
+
+{brandname} provides different `ConnectionFactory` implementations that allow you to connect to databases.
+You use JDBC connections with SQL cache stores and JDBC string-based caches stores.
+
+This section provides examples with JDBC string-based stores but the same configuration applies to SQL cache stores.
+For example, in declarative configuration you can replace the `string-keyed-jdbc-store` element with either the `table-jdbc-store` or `query-jdbc-store` elements.
+
+[discrete]
+== Connection pools
+
+Connection pools are suitable for standalone {brandname} deployments and are based on Agroal.
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/persistence_jdbc_connection_pool.xml[]
+----
+
+.ConfigurationBuilder
+[source,java,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::code_examples/PersistenceJdbcConnectionPool.java[]
+----
+
+[discrete]
+== Managed datasources
+
+Datasource connections are suitable for managed environments such as application servers.
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/persistence_jdbc_connection_datasource.xml[]
+----
+
+.ConfigurationBuilder
+[source,java,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::code_examples/PersistenceJdbcConnectionDatasource.java[]
+----
+
+[discrete]
+== Simple connections
+
+Simple connection factories create database connections on a per invocation basis and are intended for use with test or development environments only.
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
```

Comment:
shouldn't we also have YAML and JSON ?

Reply:
Yes, indeed. I plan to add that later today along with another PR for other assemblies in the Config Guide. 

Btw, @tristantarrant what do you think about tabs with "XML | JSON | YAML | ConfigurationBuilder" to show declarative alongside programmatic?

Do you think that might be too confusing? Should we separate programmatic out of the tabbed config?

Reply:
i like that